### PR TITLE
Crypto better errors in age and pbkdf2

### PR DESCRIPTION
**Pull request by semenov-vladyslav**
# Description of change

Improved error types and error handling in age and pbkdf2 modules.

## Type of change

- [x] Enhancement (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

`cargo test --all-features`

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -96,7 +96,7 @@ getrandom = { version = "0.2", optional = true, default-features = false, featur
 hmac = { version = "0.12", optional = true, default-features = false }
 lazy_static = { version = "1.4", optional = true, default-features = false }
 num-traits = { version = "0.2", optional = true, default-features = false }
-pbkdf2 = { version = "0.12", optional = true, default-features = false }
+pbkdf2 = { version = "0.12", optional = true, default-features = false, features = ["hmac"] }
```
Comment (thibault-martinez) [inline Cargo.toml]:
```suggestion
pbkdf2 = { version = "0.12", optional = true, default-features = false, features = [ "hmac" ] }
```

File: src/error.rs [hunk 2]
```diff
@@ -8,46 +8,35 @@ pub type Result<T, E = Error> = core::result::Result<T, E>;
 /// Error type of crypto.rs
 #[derive(Debug, PartialEq, Eq)]
 pub enum Error {
+    #[cfg(feature = "age")]
+    /// Age Format Error
```
Comment (thibault-martinez) [inline src/error.rs:11]: We usually put comments before attributes

File: src/error.rs [hunk 3]
```diff
@@ -8,46 +8,35 @@ pub type Result<T, E = Error> = core::result::Result<T, E>;
...
+    AgeFormatError { inner: crate::keys::age::DecError },
```
Comment (thibault-martinez) [inline src/error.rs]: Just in case: you can use `AgeFormatError(crate::keys::age::DecError)`

Comment (semenov-vladyslav) [reply src/error.rs]: Maybe I should have

Comment (thibault-martinez) [reply src/error.rs]: I don't really prefer one to the other, was just in case you were not aware of the syntax

Comment (semenov-vladyslav) [reply src/error.rs]: Thanks. I thought to keep the style of other error constructors but it's just a plain wrapper, no need for curly braces.

File: Cargo.toml [hunk 4]
```diff
@@ -62,7 +62,7 @@ wots_deprecated_do_not_use = [
 sha = [ "sha2", "digest" ]
 hmac = [ "dep:hmac" ]
 pbkdf = [ "pbkdf2" ]
```
Comment (semenov-vladyslav) [inline Cargo.toml]: Shall we require also "hmac" and "sha" deps in "pbkdf" feature? Without those features pbkdf2_hmac_sha won't work.

Comment (thibault-martinez) [reply Cargo.toml]: Yeah

File: src/keys/mod.rs [hunk 5]
```diff
@@ -5,8 +5,8 @@
 #[cfg_attr(docsrs, doc(cfg(feature = "ternary_keys")))]
 pub mod ternary;
 
-#[cfg(feature = "pbkdf")]
-#[cfg_attr(docsrs, doc(cfg(feature = "pbkdf")))]
+#[cfg(feature = "pbkdf2")]
+#[cfg_attr(docsrs, doc(cfg(feature = "pbkdf2")))]
 pub mod pbkdf;
```
Comment (thibault-martinez) [inline src/keys/mod.rs:10]: Should the module itself be pbkdf2 in case we add more pbkdf ?

Comment (thibault-martinez) [reply src/keys/mod.rs:10]: Ah I see that inside you have features for pbkdf2, ignore me

Comment (semenov-vladyslav) [reply src/keys/mod.rs:10]: PBKDF2 is a concrete algorithm, pbkdf is password-based key derivation function in general. Scrypt, for example, will nicely fit pbkdf, but not pbkdf2. If there are other pbkdfs we will declare this mod via `cfg(any(feature = "pbkdf2", feature = "scrypt"))`

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**